include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -540,7 +540,7 @@ inline void from_json(const BasicJsonType& j, std_fs::path& p)
         JSON_THROW(type_error::create(302, concat("type must be string, but is ", j.type_name()), &j));
     }
     const auto& s = *j.template get_ptr<const typename BasicJsonType::string_t*>();
-#ifdef JSON_HAS_CPP_20
+#if defined(__cpp_lib_char8_t) && (__cpp_lib_char8_t >= 201907L)
```

Comment:
Please add a comment why checking `__cpp_lib_char8_t` is required and `JSON_HAS_CPP_20` is insufficient.